src/Components/Voter.jsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from 'react';
+import * as API from '../Api';
+
+export const Voter  = ({votes, target, setVotes}) => {
+    const [vote, setVote] = useState(0);
+    const [thumb, setThumb] = useState('none');
+    const [renderCount, setRenderCount] = useState(-1)
+
+    const handleChange = (input) => {
+
+        if(thumb === 'none'){
+            if(input === 1){
+                setThumb('up');
+                setVote(() => 1);
+            }
+            if(input === -1){
+                setThumb('down');
+                setVote(() => -1);
+            }
+        }
+        if(thumb === 'up'){
+            if(input === 1){
+                setThumb('none');
+                setVote(() => -1);
+            }
+            if(input === -1){
+                setThumb('down');
+                setVote(() => -2);
+            }
+        }
+        if(thumb === 'down'){
+            if(input === 1){
+                setThumb('up');
+                setVote(() => 2);
+            }
+            if(input === -1){
+                setThumb('none');
+                setVote(() => 1);
+            }
+        }
+    
+    }
+        useEffect(() => {
+            setVotes((curr) => curr + vote);
+            setRenderCount((curr) => curr + 1)
+            target.article_id ?
+            API.patchVotes(vote,`articles/${target.article_id}`)
+            .then((data) => {
+            })
+            .catch(() => {
+                setVote((Vote) => Vote *= -1)
+                setVotes((curr) => curr - vote)
+            })
+            : API.patchVotes(vote,`comments/${target.comment_id}`)
+            .then((data) => {
+
+            })
+            .catch(() => {
+                setVote((current) => current *= -1)
+                setVotes((curr) => curr - vote)
+            })
+            }, [vote, thumb])
```

Comment:
do you need to do this inside the useEffect or could it be handled inside your handleChange function